### PR TITLE
zls: use `ReleaseSafe` instead of `ReleaseFast`

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- fast-release - slow compilation, fully optimized runtime performance, safety checks off
- safe-release - slow compilation, fully optimized runtime performance, safety checks on

relates to https://github.com/ziglang/zig/issues/288

### CODE REVIEW

File: Formula/z/zls.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def install
     else Hardware.oldest_cpu
     end
 
-    args = %W[--prefix #{prefix} -Doptimize=ReleaseFast]
+    args = %W[--prefix #{prefix} -Doptimize=ReleaseSafe]
```
Comment (carlocab) [inline Formula/z/zls.rb:32]: Seems nicer to keep `ReleaseFast` unless users report problems with it.

**Review by SMillerDev — CHANGES_REQUESTED** [review]
> relates to ziglang/zig#288

That's from 7 years ago with no relevance to Homebrew though. What are we doing this for?

**Comment by chenrui333** [comment]
> > relates to [ziglang/zig#288](https://github.com/ziglang/zig/issues/288)
> 
> That's from 7 years ago with no relevance to Homebrew though. What are we doing this for?

Oh, that is just saying the fact this option has been existing for quite some time.

**Comment by chenrui333** [comment]
also upstream is using releasesafe for their release builds though.

- https://github.com/zigtools/zls/blob/master/README.md?plain=1#L36
- https://github.com/zigtools/zls/blob/f726e9e6c9a3fc016e67e4b18e33443e7d562704/flake.nix#L44
- https://github.com/zigtools/zls/blob/f726e9e6c9a3fc016e67e4b18e33443e7d562704/.github/workflows/main.yml#L66

**Comment by SMillerDev** [comment]
Is it also the default?

**Comment by chenrui333** [comment]
> Is it also the default?

right now it is not the default in the build.zig

**Comment by chenrui333** [comment]
I think one of the reasons is that `std.builtin.Mode` is not just for release build but also debug build.

**Comment by carlocab** [comment]
> Is it also the default?

The default is always debug, IIUC.

**Comment by chenrui333** [comment]
gonna close it for now, we can revisit it later.

**Pull request closed without merging**